gcloud/storage/test_batch.py

```diff
@@ -354,39 +366,32 @@ def test_finish_nonempty_non_multipart_response(self):
         batch._requests.append(('DELETE', URL, {}, None))
         self.assertRaises(ValueError, batch.finish)
 
-    def test_current(self):
-        from gcloud.storage.batch import _BATCHES
-        klass = self._getTargetClass()
-        batch_top = object()
-        self.assertEqual(list(_BATCHES), [])
-        _BATCHES.push(batch_top)
-        self.assertTrue(klass.current() is batch_top)
-        _BATCHES.pop()
-        self.assertEqual(list(_BATCHES), [])
-
     def test_as_context_mgr_wo_error(self):
-        from gcloud.storage.batch import _BATCHES
+        from gcloud.storage.client import Client
         URL = 'http://example.com/api'
         expected = _Response()
         expected['content-type'] = 'multipart/mixed; boundary="DEADBEEF="'
         http = _HTTP((expected, _THREE_PART_MIME_RESPONSE))
         connection = _Connection(http=http)
-        client = _Client(connection)
+        project = 'PROJECT'
+        credentials = _Credentials()
+        client = Client(project=project, credentials=credentials)
+        client._connection = connection
```

Comment:
I'm not sure what's double wrong, but the main goal is to avoid the HTTP request. You did make me realize I could just do

```python
client._connection._http = http
```

instead of making a custom connection.